### PR TITLE
Allow iam_policy to maintain idempotence if the role referenced has …

### PR DESCRIPTION
**Pull request by wimnat**
…been removed

As per conversation - https://groups.google.com/forum/#!topic/ansible-devel/hseexgcjnUs

@defionscode

### CODE REVIEW

**Comment by robynbergeron** [comment]
[module: cloud/amazon/iam_policy.py]

**Comment by robynbergeron** [comment]
Hi @wimnat -- thanks for this PR, as always :) /me bows in gratitude :bow: 

@defionscode -- Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please add a comment with "shipit" in the text, and we will flag it for inclusion.

If it fails these guidelines, please add a comment with "needs_revision" in the text, along with the needed changes, and we will flag it for revision.

File: cloud/amazon/iam_policy.py [hunk 1]
```diff
@@ -183,6 +183,14 @@ def role_action(module, iam, name, policy_name, skip, pdoc, state):
     current_policies = [cp for cp in iam.list_role_policies(name).
                                         list_role_policies_result.
                                         policy_names]
+  except boto.exception.BotoServerError as e:
+    if e.error_code == "NoSuchEntity":
+      # Role doesn't exist so it's safe to assume the policy doesn't either
```
Comment (juliedavila) [inline cloud/amazon/iam_policy.py:188]: What do you think about doing a check for the policy here? Isn't it possible that the policy exists even if the role doesn't?

Comment (tcable) [inline cloud/amazon/iam_policy.py:188]: If the role doesn't exist then there can be no managed policies attached to it and no inline policies will exist. Also the current version of Boto (2.38.0) can't check for managed policies on a role, although this functionality is currently being added (develop branch of Boto).

**Comment by gregdek** [comment]
@defionscode back in your court. Let us know if it's "shipit" or "needs_revision". Thanks!

File: cloud/amazon/iam_policy.py (hunk 1 above)
Comment (wimnat) [inline cloud/amazon/iam_policy.py:188]: @defionscode - Yep, what @tcable said.  An inline policy can not exist without the role.

**Comment by juliedavila** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @wimnat for the PR and @defionscode for the review. Marking for potential inclusion.

**Pull request merged**